### ISSUE TITLE
Incorrect standard driver for `@inbook`

### ISSUE THREAD

**Issue by Skillmon** [plk/biblatex#943]
The standard driver for `@inbook` doesn't take into account that the (according to the documentation) optional key `bookauthor` might not be set. So with the standard.bbx driver one gets an empty block instead of the `bookauthor` formatted like this: `<author>, <title>, in: , <booktitle>, <editor>` (if an editor was given).

**Comment by moewew** [comment]
Can you please share an MWE for that issue?

```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=numeric, backend=biber]{biblatex}

\begin{filecontents}[force]{\jobname.bib}
@inbook{appleby,
  author    = {Humphrey Appleby},
  title     = {On the Importance of the Civil Service},
  booktitle = {Why We Matter},
  editor    = {Bernard Woolley},
  date      = {1980},
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
\cite{appleby}
\printbibliography
\end{document}
```
seems to look just fine for me.
> Humphrey Appleby. ‘On the Importance of the Civil Service’. In: Why We Matter. Ed. by Bernard Woolley. 1980.

**Comment by Skillmon** [comment]
The cause is a redefinition of `\newunitpunct` I did a while ago...

```latex
\documentclass[]{article}

\usepackage[ngerman]{babel}
\usepackage[]{csquotes}
\usepackage[style=numeric,backend=biber]{biblatex}

\begin{filecontents}[force]{\jobname.bib}
@Inbook{Miikkulainen2010,
author="Miikkulainen, Risto",
editor="Sammut, Claude
and Webb, Geoffrey I.",
title="Topology of a Neural Network",
bookTitle="Encyclopedia of Machine Learning",
year="2010",
publisher="Springer US",
address="Boston, MA",
isbn="978-0-387-30164-8",
doi="10.1007/978-0-387-30164-8_837",
url="https://doi.org/10.1007/978-0-387-30164-8_837"
}
\end{filecontents}
\addbibresource{\jobname.bib}

\renewcommand*\newunitpunct{, }

\begin{document}
\cite{Miikkulainen2010}
\printbibliography
\end{document}
```

What would be the correct approach to set `,` as the unit delimiter instead of `.`?

**Comment by moewew** [comment]
```latex
\renewcommand*\newunitpunct{\addcomma\space}
```
would be correct. But that only works since `\add...` does a *lot* of work behind the scenes to avoid double punctuation.

The real culprit is.
https://github.com/plk/biblatex/blob/ce59568b265b8bb2e079029c5caf8e9ee7081266/tex/latex/biblatex/biblatex.def#L2863-L2865
The macro prints the punctuation directly in `\printtext` and not in a `\setunit` or `\printunit` as usual.

https://tex.stackexchange.com/q/199173/35864 and on https://tex.stackexchange.com/q/175730/35864 TeX.SX are about the same issue (also https://tex.stackexchange.com/q/350872/35864, https://tex.stackexchange.com/q/393034/35864). See also https://github.com/michal-h21/biblatex-iso690/issues/21.

**Comment by Skillmon** [comment]
I've closed this issue as it could have been avoided if I read the documentation correctly :) Thanks for the fast replies! Do you intend to change the `in:` bibmacro in the future?

**Comment by moewew** [comment]
It's a tough one.

I believe that `\add...` punctuation commands should exclusively be wrapped in `\setunit`/`\printunit` or appear in `\DeclareFieldFormat` and friends (but then *not* in trailing/final position). As such
```latex
 \newbibmacro*{in:}{% 
   \printtext{% 
     \bibstring{in}\intitlepunct}} 
```
looks wrong to me and I would much prefer
```latex
\newbibmacro*{in:}{% 
  \printtext{\bibstring{in}}%
  \printunit{\intitlepunct}}
```
But this means we have to deploy `\printunit` (since `\setunit` would get overwritten by subsequent `\setunit` calls in cases like this where the `bookauthor` is missing). Since `\printunit` overrides all following `\setunit`s we'd have to look very carefully at the following punctuation commands and check that the output for 'sparse' entries matches our expectations. (A similar issue arose in https://github.com/plk/biblatex/pull/557. There is a good reason why there are only 7 uses of `\printunit` in all of the standard styles.)

I'll reopen this so we can think about this for/in a while.

**Comment by moewew** [comment]
In this particular case the issue could also be resolved by choosing the entry type `@incollection` (or maybe even `@inreference`), which seems more fitting than ` @inbook` here.
```latex
\documentclass{article}
\usepackage[ngerman]{babel}
\usepackage{csquotes}
\usepackage[style=numeric,backend=biber]{biblatex}

\renewcommand*\newunitpunct{, }
%\renewcommand*\newunitpunct{\addcomma\space}

\begin{filecontents}[force]{\jobname.bib}
@incollection{Miikkulainen2010,
  author    = {Miikkulainen, Risto},
  editor    = {Sammut, Claude and Webb, Geoffrey I.},
  title     = {Topology of a Neural Network},
  booktitle = {Encyclopedia of Machine Learning},
  year      = {2010},
  publisher = {Springer US},
  address   = {Boston, MA},
  isbn      = {978-0-387-30164-8},
  doi       = {10.1007/978-0-387-30164-8_837},
}
\end{filecontents}
\addbibresource{\jobname.bib}

\begin{document}
\cite{Miikkulainen2010}
\printbibliography
\end{document}
```
In theory every `@inbook` should have a `bookauthor` in much the same way every `@book` should have an `author`. (Of course there *are* exceptions when a `@book` may not have an `author`.) So if you are missing the `bookauthor` you are almost always looking at an `@incollection` instead of an `@inbook`.

But of course the definition of `in:` is still problematic.

**Comment by Skillmon** [comment]
That was just a direct citation export from some site, I haven't given it any thought other than that I noticed the doubled comma in the output. However everything is displayed correctly now, so I won't change the entry. Thanks again!

**Comment by moewew** [comment]
I think we should bite the bullet and go for 
```latex
\newbibmacro*{in:}{%
  \bibstring{in}%
  \printunit{\intitlepunct}}
```
I merged #944 with the required changes. If my thinking is correct there shouldn't be any side effects to this except for the obvious bug fix for https://tex.stackexchange.com/q/175730/ and the issue described here.

People patching `in:` might experience problems with this change, but I hope that people don't patch `in:` that much because it is so short.

**Comment by Skillmon** [comment]
@moewew I think to change `in:` the go-to method is to set the bibstring, not to redefine the bibmacro, so most people should have done that rather than a redefinition or patch.
